tests/DataSources/ProjectsTest.php

```diff
@@ -4,56 +4,209 @@
 
 namespace Doctrine\Website\Tests\DataSources;
 
-use Doctrine\Website\DataBuilder\ProjectDataBuilder;
-use Doctrine\Website\DataBuilder\WebsiteData;
-use Doctrine\Website\DataBuilder\WebsiteDataReader;
+use DateTimeImmutable;
 use Doctrine\Website\DataSources\Projects;
+use Doctrine\Website\Docs\RST\RSTLanguage;
+use Doctrine\Website\Docs\RST\RSTLanguagesDetector;
+use Doctrine\Website\Git\Tag;
+use Doctrine\Website\Projects\GetProjectPackagistData;
+use Doctrine\Website\Projects\ProjectDataReader;
+use Doctrine\Website\Projects\ProjectDataRepository;
+use Doctrine\Website\Projects\ProjectGitSyncer;
+use Doctrine\Website\Projects\ProjectVersionsReader;
 use Doctrine\Website\Tests\TestCase;
 use PHPUnit\Framework\MockObject\MockObject;
 
 class ProjectsTest extends TestCase
 {
-    private WebsiteDataReader&MockObject $dataReader;
+    private ProjectDataRepository&MockObject $projectDataRepository;
 
-    private Projects $projects;
+    private ProjectGitSyncer&MockObject $projectGitSyncer;
 
-    protected function setUp(): void
-    {
-        $this->dataReader = $this->createMock(WebsiteDataReader::class);
+    private ProjectDataReader&MockObject $projectDataReader;
 
-        $this->projects = new Projects(
-            $this->dataReader,
-        );
-    }
+    private ProjectVersionsReader&MockObject $projectVersionsReader;
+
+    private RSTLanguagesDetector&MockObject $rstLanguagesDetector;
+
+    private GetProjectPackagistData&MockObject $getProjectPackagistData;
+
+    private string $projectsDir;
 
-    public function testGetSourceRows(): void
+    private Projects $dataSource;
+
+    public function testBuild(): void
     {
+        $this->projectDataRepository->expects(self::once())
+            ->method('getProjectRepositoryNames')
+            ->willReturn(['orm']);
+
+        $this->projectGitSyncer->expects(self::any())
+            ->method('checkoutDefaultBranch')
+            ->with('orm');
```

Comment:
Does this expectation actually expect anything? In other words, when could it fail?

Reply:
I took the code from `lib/DataBuilder/ProjectDataBuilder.php` and with it also the old tests. I should review the test and make some useful expectations.

Reply:
Commit [b05d808e4373c3eb29b6c756d5d1c9a780f78f4a](https://github.com/doctrine/doctrine-website/pull/592/commits/b05d808e4373c3eb29b6c756d5d1c9a780f78f4a) does fix the expected number of calls, but it doesn't appear in the PR :thinking:

Reply:
I see it just fine.